### PR TITLE
Adding Artifactory APK doc

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This PR adds a doc on using Artifactory as a pull through cache for chainguard's package repositories.

### What should this PR do?

resolves https://github.com/chainguard-dev/internal/issues/4387

### Why are we making this change?

Sales request from @ryryryano 

I/We will also need to update this doc in the nearish future once the language ecosystem package repositories are available (see https://github.com/chainguard-dev/internal/issues/4420)

### How should this PR be tested?

You may need to set up an artifactory instance to test it out. I can hook you up with mine if you'd like.

Note: it's a pain to go through this if your email is your Artifactory username. Again, let me know if I can help!

### CODE REVIEW

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+```shell
+docker build -t ar-virtual -f Dockerfile.virtual .
+```
+
+This command will install the `nodejs-18` package from the virtual repository as it builds the image.
+
+
+## Debugging pull through from Chainguard’s registry to Artifactory
+
+If you run into issues when trying to pull from Chainguard's package repositories through Artifactory, you can try checking these common pitfalls:
```
Comment (smythp) [inline content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md:231]:
```suggestion
If you run into issues when trying to pull from Chainguard's package repositories through Artifactory, you can try checking for these common pitfalls:
```

**Review by smythp — APPROVED** [review]
Good in-depth tutorial. Testing everything was tricky, replacing everything is pretty failure-prone (not sure if there's a way around that), and I'm especially well set up for doing this kind of replacement. I made a suggestion about using env variables at the top to make it easier for users, but now I'm thinking given how many Dockerfiles are created, changing things around is likely to cause problems, so perhaps just keep things the way they are.

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+RUN echo 'https://<ARTIFACTORY-USER-PROFILE>:<TOKEN>@<ARTIFACTORY-HOSTNAME>.jfrog.io/artifactory/cg-wolfi/' >> /etc/apk/repositories
+RUN echo 'https://<ARTIFACTORY-USER-PROFILE>:<TOKEN>@<ARTIFACTORY-HOSTNAME>.jfrog.io/artifactory/cg-extras/' >> /etc/apk/repositories
+RUN apk add nodejs-22
+USER nonroot
+EOF
+```
+
+This example will install the `nodejs-22` package into the built image.
+
+After creating the Dockefile, build the image. Note that this command includes the `-f` option to specify that it should build the image with the Dockerfile you just created:
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md:174]: good catch, I actually found another identical spelling error!

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+## Prerequisites
+
+In order to complete this tutorial, you will need administrative privileges over an Artifactory instance. If you're interested in testing out this configuration, you can set up a trial instance on the [JFrog Artifactory landing page](https://jfrog.com/artifactory/). 
+
+> **Note**: As you follow this guide, you may run into issues if your Artifactory username is an email address; specifically, the `@` sign can lead to errors. Be sure that you're using a user profile with a name that only contains letters and numbers. 
+
+## Setting Up Remote Artifactory Repositories to Mirror Public Chainguard Package Repositories
+
+To set up a remote repository in Artifactory from which you can pull public Chainguard packages, log in to the JFrog platform. Once there, click on the **Administration** tab near the top of the screen and select **Repositories** from the left-hand navigation menu. On the Repositories page, click the **Create a Repository** button and select the **Remote** option.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md:33]: that's a great point and something I hadn't considered, thanks for flagging

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-images-pull-through/index.md [hunk 4]
```diff
@@ -1,9 +1,10 @@
 ---
 title: "How to Set Up Pull Through from Chainguard Registry to Artifactory"
-linktitle: "Artifactory"
+linktitle: "Images"
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-images-pull-through/index.md:3]: Love it, thanks. I could sense these linktitles weren't ideal

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-images-pull-through/index.md [hunk 5]
```diff
@@ -15,7 +16,7 @@ menu:
   docs:
     parent: "pull-through-guides"
 toc: true
-weight: 010
+weight: 005
 ---
 
 Organizations can use Chainguard Images along with third-party software repositories in order to integrate with current workflows as the single source of truth for software artifacts. In this situation, you can set up a remote repository to function as a mirror of a [Chainguard Registry](/chainguard/chainguard-registry/overview/) — either the public registry or a private one belonging to your organization. This mirror can then serve as a pull through cache for your Chainguard Images.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-images-pull-through/index.md:22]: totally fair. I have a habit of writing an intro paragraph that doesn't really add anything

File: content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+Take note of the token generated for each repository before moving on.
+
+### Testing pull through from remote package repositories interactively
+
+In order to test that you're able to pull packages from the remote Artifactory repositories you just created, this section outlines how to build a container image (using a Chainguard image as a base) and configuring it to pull packages from these repositories.
+
+Open up a terminal and create a Dockerfile:
+
+```shell
+cat > Dockerfile <<EOF
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md:80]: you're totally right, this is a much cleaner solution. Thanks for hammering it out!

**Pull request merged**